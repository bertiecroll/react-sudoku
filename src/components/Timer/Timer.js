import React, { Component } from 'react'
import { displayCount } from 'lib/formatters'

import './Timer.css'

export default class Timer extends Component {

  state = {
    elapsedTime: 0,
    previousTime: Date.now(),
  }

  onTick = () => {
    const now = Date.now()
    this.setState({
      elapsedTime: (this.state.elapsedTime + (now - this.state.previousTime)),
      previousTime: Date.now(),
    })
  }

  componentDidMount () {
    this.interval = setInterval(this.onTick)
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render () {
    const seconds = Math.floor(this.state.elapsedTime / 1000) % 60
    const minutes = Math.floor(this.state.elapsedTime / 1000 / 60)
    return (
      <div className="Timer">
        Timer: {displayCount(minutes)}:{displayCount(seconds)}
      </div>
    )
  }
}
