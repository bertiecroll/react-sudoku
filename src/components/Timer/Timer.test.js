import React from 'react'
import { shallow } from 'enzyme'
import Timer from 'components/Timer'

describe('Timer', () => {

  it('renders div with timer count', () => {
    const wrapper = shallow(<Timer />)
    expect(wrapper).toMatchSnapshot()
  })
})
