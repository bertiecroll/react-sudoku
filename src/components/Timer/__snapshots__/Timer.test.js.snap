// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer renders div with timer count 1`] = `
<div
  className="Timer"
>
  Timer: 
  00
  :
  00
</div>
`;
